src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/2.sql

```diff
@@ -708,6 +775,181 @@ AS
         AND StudyKey = @studyKey
     END
 
+    -- Insert Custom Tags
+    DECLARE @rowCount AS BIGINT
+    -- This variable is used for not returning value from SELECT.
+    DECLARE @temp AS INT
+
+    -- String Key tags
+    SET @rowCount = (SELECT COUNT(TagKey) FROM @stringCustomTags)
+    IF @rowCount <> 0
+    BEGIN
+        -- Validate and lock custom tags
+        SELECT @temp = dbo.CustomTag.TagKey
+        FROM dbo.CustomTag WITH (UPDLOCK)
```

Comment:
You don't need this select.

Instead change the MERGE condition to add only those customTag whose status is 2.

Reply:
hint it with repeatableread